test/unit/fact_value_test.rb

```diff
@@ -121,41 +121,60 @@ def setup
       end
     end
 
-    test "only return facts from host in user's taxonomies" do
-      user_role = FactoryGirl.create(:user_user_role)
-      FactoryGirl.create(:filter, :role => user_role.role, :permissions => Permission.where(:name => 'view_hosts'), :search => "hostgroup_id = #{target_host.hostgroup_id}")
+    context 'taxonomies' do
+      setup do
+        @orgs = FactoryGirl.create_pair(:organization)
+        @locs = FactoryGirl.create_pair(:location)
+      end
 
-      orgs = FactoryGirl.create_pair(:organization)
-      locs = FactoryGirl.create_pair(:location)
-      target_host.update_attributes(:location => locs.last, :organization => orgs.last)
+      context 'limited view permissions' do
+        setup do
+          setup_user('view', 'hosts',
```

Comment:
In response to https://github.com/theforeman/foreman/pull/2870/files#r47493301 , just changing the order and creating `@orgs` `@locs`first fixed the issue. 

If I call `setup_user` first, then create an organization, it gets added to the user automatically, which I didn't want in this case.